how-to/misc/identity-provider-enablement.rst

```diff
@@ -0,0 +1,354 @@
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+For the purpose of this document we will assume you have a model called `iam` which contains a deployment of Canonical Identity Platform.
+Integration is done by consuming an offer for the `oauth` endpoint of the `Hydra <https://charmhub.io/hydra>`_ charm. If your
+identity platform uses a custom CA, you also need to consume the `send-ca-cert` endpoint deployed in your `iam` model.
+
+Create the offers:
+
+::
+
+    juju offer iam.hydra:oauth
+    juju offer iam.self-signed-certificates:send-ca-cert
+
+Get the offer URLs. We'll need them when creating the config file:
+
+::
+
+    HYDRA_OFFER=$(juju list-offers --format=json -m iam hydra | jq -r '.hydra.["offer-url"]')
+    SEND_CA_CERT_OFFER=$(
+        juju list-offers \
+        --format=json \
+        -m iam \
+        self-signed-certificates | jq -r '.["self-signed-certificates"].["offer-url"]')
+
+Create the config file:
+
+::
+
+    cat << EOF > canonical-iam.yaml
+    oauth_offer: $HYDRA_OFFER
+    cert_offer: $SEND_CA_CERT_OFFER
+    EOF
+
+And finally add the provider:
+
+::
+
+    sunbeam identity provider add \
+        canonical openid canonical-identity-platform \
+        --config canonical-iam.yaml
+
+In the above example we added a new identity provider of type `canonical` with the `openid` protocol and the name `canonical-identity-platform`.
+the name is important in the case of providers of type `canonical` as the name is also used as a label in `Horizon` when selecting the IDP. So make
+sure to use a descriptive name.
+
+Now we can list the providers:
+
+::
+
+    sunbeam identity provider list
+    +-----------------------------+------------+----------+
+    | Name                        | Provider   | Protocol |
+    +-----------------------------+------------+----------+
+    | Keystone Credentials        | Built-in   | keystone |
+    | canonical-identity-platform | canonical  | openid   |
+    +-----------------------------+------------+----------+
+
+At this point the configurations exist in apache, keystone and horizon that enable a cloud administrator to configure it as an authentication backend. We will
+cover that part later.
+
+External IDPs
+~~~~~~~~~~~~~
+
+Sunbeam currently supports the following IDP types:
+
+* `Google <https://developers.google.com/identity/openid-connect/openid-connect>`_
+* `Okta <https://help.okta.com/en-us/content/topics/apps/apps_app_integration_wizard_oidc.htm>`_
+* `Entra ID <https://learn.microsoft.com/en-us/entra/identity-platform/v2-protocols-oidc#enable-id-tokens>`_
+
+Each of these provider types require specific configurations in order to enable them and each have their own procedure for creating the client credentials needed to initiate the authentication workflow.
+You will have to consult the official documentation for each, in order to generate the required client credentials. The configuration formats that Sunbeam requires are detailed bellow. The configuration
```

Comment:
In general, this is sunbeam for sure - but the official name of the product should be used in this documentation - Canonical OpenStack. As such, s/Sunbeam/Canonical OpenStack/g